src/cpp/src/visual_language/pipeline.cpp

```diff
@@ -93,19 +93,22 @@ class ov::genai::VLMPipeline::VLMPipelineImpl {
 
         ov::Tensor inputs_embeds = m_inputs_embedder->get_inputs_embeds(prompt, rgbs);
 
-        Sampler sampler = Sampler(m_tokenizer);
-
         std::vector<SequenceGroup::Ptr> requests;
         size_t request_id = 0;
         size_t block_size = 1; // not used
         bool enable_prefix_caching = false;
         size_t history_size = m_language.get_tensor("attention_mask").get_shape().at(1);
         size_t inputs_embeds_size = inputs_embeds.get_shape().at(1);
-        ov::Tensor prompt_ids(ov::element::i64, { history_size + inputs_embeds_size });
-        std::fill_n(prompt_ids.data<int64_t>(), prompt_ids.get_size(), 0);
+
+        auto chat_history = m_inputs_embedder->get_tokenized_chat_history();
+        size_t chat_history_size = std::max(chat_history.get_shape().at(1), history_size + inputs_embeds_size);
```

Comment:
not clear why do we need max here?
because history can contain only special image tokens, which is converted to more values in inputs_embeds ? In this case, `history_size + inputs_embeds_size` looks always correct.

IMO, `get_tokenized_chat_history` does not provide actual information about history size

Reply:
for tiny-random-minicpmv-2_6 tokenized history is bigger than attention mask + new inputs
not sure that it should be like this, just a practical note